File: Prudhvi-LLD-Mar/BehavioralDesignPattern/WeatherStationObserverDesignPattern/CurrentConditionsDisplay.java
package WeatherStationObserverDesignPattern;

import java.text.MessageFormat;

public class CurrentConditionsDisplay implements Display {

  private double currentTemperature;
  private double currentHumidity;

  @Override
  public void update(WeatherStationMessage weatherStationMessage) {
    this.currentTemperature = weatherStationMessage.getTemperature();
    this.currentHumidity = weatherStationMessage.getHumidity();
  }

  @Override
  public String display() {
    return MessageFormat.format("Current conditions: {0}°C and {1}% humidity.",
        currentTemperature, currentHumidity);
  }
}

File: rahul-narang-LLD-March/src/CurrentConditionsDisplay.kt
class CurrentConditionsDisplay(
    var name: String
) : Observer {
    private var temp: Double = 0.0
    private var humidity: Double = 0.0

    override fun update(weatherStationMessage: WeatherStationMessage) {
        this.temp = weatherStationMessage.temp
        this.humidity = weatherStationMessage.humidity
    }

    fun getCurrentWeatherConditions(): String {
        return "Current Conditions: ${this.temp} degree C, ${this.humidity} % Humidity"
    }
}